src/frontend/komponenter/Felleskomponenter/HeaderMedSøk/OpprettFagsakModal.tsx

```diff
@@ -55,16 +61,23 @@ const OpprettFagsakModal: React.FC<IOpprettFagsakModal> = ({
     const visModal = !!søkeresultat || !!personInfo;
     const [fagsakType, settFagsakType] = useState<FagsakType>(FagsakType.NORMAL);
     const [visFeilmelding, settVisFeilmelding] = useState(false);
-    const [valgtInstitusjon, settValgtInstitusjon] = useState<IInstitusjon | undefined>(undefined);
+    const [valgtSamhandler, settValgtSamhandler] = useState<ISamhandlerInfo | undefined>(undefined);
+    const { onSubmitWrapper, samhandlerSkjema } = useSamhandlerSkjema();
 
-    const validerInput = () => {
-        return fagsakType === FagsakType.INSTITUSJON ? !!valgtInstitusjon?.navn : true;
-    };
     const onClose = () => {
         settFagsakType(FagsakType.NORMAL);
         settVisFeilmelding(false);
+        settValgtSamhandler(undefined);
         lukkModal();
     };
+
+    useEffect(() => {
+        if (samhandlerSkjema.submitRessurs.status === RessursStatus.SUKSESS) {
+            settValgtSamhandler(samhandlerSkjema.submitRessurs.data);
+            settVisFeilmelding(false);
+        }
+    }, [samhandlerSkjema.submitRessurs.status]);
```

Comment:
`samhandlerSkjema.submitRessurs.data` må også med i listen her, da den brukes i effekten

Reply:
Det fikk jeg ikke lov til å gjøre pga. data ikke nødvendigvis finnes:
`TS2339: Property 'data' does not exist on type 'Ressurs<ISamhandlerInfo>'.   Property 'data' does not exist on type '{ status: RessursStatus.IKKE_HENTET; }'.`
Men 'data' kan vel uansett ikke endre seg uten en midlertidig endring av status til RessursStatus.HENTER i mellomtiden? I så fall burde vi være dekket

Reply:
Jeg sjekket koden der hvor submitRessurs.status settes i familie-felles-frontend for ordens skyld, og der settes den til HENTER først, før request-kallet som oppdaterer data 👍